autosar/portinterface.py

```diff
@@ -119,8 +119,11 @@ def append(self,elem):
         """
         adds elem to the self.parameters list and sets elem.parent to self (the port interface)
         """
-        if not isinstance(elem, autosar.element.ParameterDataPrototype):
-            raise ValueError("Expected elem variable to be of type ParameterDataPrototype")
+        if (
+            not isinstance(elem, autosar.element.AutosarDataPrototype)
+            or elem.role is not autosar.element.AutosarDataPrototype.Role.Parameter
+        ):
+            raise ValueError("Expected elem variable to be of type AutosarDataPrototype (Parameter)")

```

Comment:
Probably can simplify this with the following

```suggestion
        if not isinstance(elem, AutosarDataPrototype) or elem.role is not AutosarDataPrototype.Role.Parameter:
            raise ValueError("Expected elem variable to be of type AutosarDataPrototype (Parameter)")

```